### PR TITLE
update github template and CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by samir-nasibli**
# Description
I am proposing some changes on our github templates for PRs.
let's discuss unification of our PRs.
Please propose your suggestions for the each section.

Based on https://github.com/oneapi-src/oneTBB/blob/master/.github/pull_request_template.md

### CODE REVIEW

**Review by Alexsandruss — CHANGES_REQUESTED** [review]

File: .github/Pull_Request_template.md [hunk 1]
```diff
@@ -1,9 +1,47 @@
-# Description
+### Description 
+_Add a comprehensive description of proposed changes_
```
Comment (alexandraepan) [inline .github/Pull_Request_template.md:2]: we can delete this line

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -40,6 +40,20 @@ To contribute your changes directly to the repository, do the following:
 - Document your code.
 - [Submit](https://github.com/intel/scikit-learn-intelex/pulls) a pull request into the `master` branch. Provide a brief description of the changes you are contributing.
 
+Standard acronyms to start the commit message with are
+API: an (incompatible) API change
```
[the remark this answers is not in the document]
Comment (samir-nasibli) [reply CONTRIBUTING.md:44]: This acronyms taken from https://numpy.org/doc/stable/dev/development_workflow.html#writing-the-commit-message.
This is for features/enhancements that brings API changes, that also could break compatibility.

We may discuss on the set of acronyms used for git messages.

**Comment by samir-nasibli** [comment]
I suggest not to pile up the PRs template. I will collect your feedbacks and comments. I think a lot them can be transferred to [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/pull/1388/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055).
In the template itself, we can leave a link to CONTRIBUTING.md and leave the main checklists useful code owners and reviewers

**Comment by samir-nasibli** [comment]
I will move PR template changes into separate topic.

**Comment by ethanglaser** [comment]
Closing due to lack of recent activity - please re-open when progress continues

**Comment by samir-nasibli** [comment]
@aepanchi @emmwalsh Please, take a look, and in case please close this PR.

**Pull request closed without merging**